lib/StripeResource.js

```diff
@@ -192,14 +197,49 @@ StripeResource.prototype = {
       callback.call(
         self,
         new Error.StripeConnectionError({
-          message: 'An error occurred with our connection to Stripe',
+          message: self._generateConnectionErrorMessage(requestRetries),
```

Comment:
nit, let's just inline this since it's a single expr anyway.

Reply:
This makes for easier testing though as @robz-stripe mentioned in an earlier comment.

Reply:
Ah righto, sgtm